iree/compiler/InputConversion/MHLO/ConvertMHLOToLinalgExt.cpp

```diff
@@ -161,25 +167,55 @@ struct SortOpConversion : public OpConversionPattern<mhlo::SortOp> {
   using OpConversionPattern<mhlo::SortOp>::OpConversionPattern;
 
   LogicalResult matchAndRewrite(
-      mhlo::SortOp mhloSortOp, OpAdaptor adaptor,
+      mhlo::SortOp op, OpAdaptor adaptor,
       ConversionPatternRewriter &rewriter) const final {
+    Location loc = op.getLoc();
+    llvm::SmallVector<Value> inputs;
+    for (auto input : adaptor.getOperands()) {
+      Type newType = convertShapedToSignless(input.getType());
+      if (newType == input.getType()) {
+        inputs.push_back(input);
+      } else {
+        inputs.push_back(
```

Comment:
Doesnt the dialect conversion automatically insert the unrealized conversion casts?

Reply:
I always struggle with getting the dialect conversion to do what I want. Managed to get it functional.